numpy-stubs/__init__.pyi

```diff
@@ -575,6 +590,10 @@ def asarray(a: ndarray, dtype: Type[float]) -> ndarray[float64]: ...
 @overload
 def asarray(a: ndarray, dtype: Type[str]) -> ndarray[str_]: ...
 @overload
+def atleast_2d(ary: _Coercable) -> ndarray: ...
+@overload
+def atleast_2d(ar: _Coercable, *ary: _Coercable) -> List[ndarray]: ...
```

Comment:
I'm guessing it didn't work with just `*ary`?

Reply:
No, I think Mypy must treat `*arr` as 1 or more `arr` because it couldn’t distinguish between the 2 signatures. I say this is Mypy behaviour as I would’ve thought `*arr` would mean 0 or more, but that might be me getting it wrong.

Reply:
I think the problem is just that the  return types are incompatible. Like, I'm pretty sure this

```python
@overload
def f(ary: int) -> Tuple[int]: ...
@overload
def f(*ary: int) -> Tuple[int, ...]: ...
```
would have worked because `Tuple[int]` is a subclass of `Tuple[int, ...]`.

The reasoning by mypy is this: if the first overload is a special case of the second, then the return type of the first should also be a special case of the return type of the second. But `ndarray` is not a special case of `List[ndarray]`, so it didn't work.

The reasons behind that behavior of mypy are explained here: https://mypy.readthedocs.io/en/stable/more_types.html#type-checking-the-variants